pkg/services/internal/whatsonchain/service.go

```
@@ -165,3 +166,46 @@ func (woc *WhatsOnChain) FindChainTipHeader(ctx context.Context) (*wdk.ChainBloc
 
 	return header, nil
 }
+
+// PostBEEF attempts to post beef with given txIDs
+func (woc *WhatsOnChain) PostBEEF(ctx context.Context, beef *transaction.Beef, txIDs []string) (*wdk.PostedBEEF, error) {
+	if len(txIDs) == 0 {
+		return nil, fmt.Errorf("no txids provided")
+	}
+
+	if beef == nil {
+		return nil, fmt.Errorf("beef is required to post transactions")
+	}
+
+	txResults := make([]wdk.PostedTxID, 0, len(txIDs))
+
+	for _, txid := range txIDs {
+		tx := beef.FindTransaction(txid)
+		if tx == nil {
+			return nil, fmt.Errorf("cannot find transaction for txid: %s", txid)
+		}
+
+		rawTxBytes := tx.Bytes()
+		if len(rawTxBytes) == 0 {
+			return nil, fmt.Errorf("empty raw transaction bytes for txid: %s", txid)
+		}
+
+		broadcastResult, err := woc.Broadcast(ctx, rawTxBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to broadcast transaction %s: %w", txid, err)
```

we shouldn't return error here.
The expected result of PostBEEF is a result for each transaction even if it was an error.

The only case when we should return an error, is when the provided arguments are invalid.

So in case when woc.Broadcast returns an error, we should append txResults with PostedTxID error result.